### PR TITLE
Joss Responses: Library Documentation and Basic Profiling

### PR DESCRIPTION
**Pull request by AlexBuccheri**
This MR adds documentation on the functionality of the library (purpose and usage), and adds basic benchmarking:

I've enabled Github pages for this project. New pages should be included in `/docs` of the GX repo. I suggest following this [repo](https://github.com/kbroman/simple_site/blob/gh-pages/index.md) as a guide, and if we need to, add [Jekyll]( https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll) support.

The webpage will be: nomad-coe.github.io

TODOs:
- [x] Add performance plots to Github pages
    - Tentatively assigned to @Panadestein  
    -  This may also require adding something to the CI... I didn't read the docs in any detail. But it can be looked at once the markdown is added.
- [x] Add library description to Time-frequency README, drawing on notes from our papers, and CP2K usage
    - Assigned to @AlexBuccheri (although happy to take any input)
- [ ] Add an example of calling the API to do a) forward transform and b) backward transform to the time-frequency README
    - Preferable for one of the current lib users (ABINIT and AIMS) to add this

Contributions from all developers welcome

### CODE REVIEW

File: JOSS/paper.md [hunk 1]
```diff
@@ -149,58 +149,57 @@ Our library requires as input the grid size $n$, the minimal eigenvalue differen
...
-│   ├── test
-│   │   ├── conftest.py
-│   │   ├── test_gx_minimax_grid.f90
-│   │   ├── test_gx_minimax_grid.py
-│   │   └── test_gx_tabulate_minimax.py
-│   └── utilities
-│       └── gx_tabulate_minimax.F90
-├── LICENSE.txt
-└── README.md
+```plaintext
```
Comment (AlexBuccheri) [inline JOSS/paper.md:168]: I note that I changed the symbols as the unicode prevented local compilation to latex.

However, I recommend removing this diagram as it takes up alot of space in an already-tight document, and will quickly become out-of-date if any source structure changes.

File: JOSS/refs.bib [hunk 2]
```diff
@@ -1,83 +1,61 @@
-@article{wang2021assessing,
```
Comment (AlexBuccheri) [inline JOSS/refs.bib:1]: wang2021assessing and ren2015beyond were duplicates.

All other changes are either auto-formatting or unicode correction.

File: JOSS/paper.md (hunk 1 above)
Comment (aziziph) [reply JOSS/paper.md:168]: @dgolze If I remember correctly, we were discussing this diagram and Dorothea was in favor of keeping the diagram.

**Review by JWilhelm — APPROVED** [review]

**Comment by AlexBuccheri** [comment]
@JWilhelm thanks for approving but it would be preferable to address the TODOs in the description prior to merging. As the tasks are relatively non-overlapping, I think it's fine to work in the same branch

**Comment by Panadestein** [comment]
Hey @AlexBuccheri I have added a section with the CH4 benchmark (verbatim from the issue, it is just md). Now for the action, I think GitHub deals with that alone, you just have to enable it in the Settings of the repo. Since we have only md files, Jekyll will do that just fine. I have some actions for my websites, but only because I use other formats and I convert the files to HTML (basically the same as Jekyll does).

**Comment by Panadestein** [comment]
@AlexBuccheri @fdelesma @JWilhelm @aziziph @gmatteo 

Proposal from @dgolze for the brief repo description:

"Library for Green’s function based electronic structure theory calculations"

or

"Tools for Green’s function based electronic structure theory calculations"

Let's iterate this a bit.

**Review by dgolze — APPROVED** [review]

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+[Markdown](https://daringfireball.net/projects/markdown/),
+[GitHub](https://github.com) hosts the site for you, and you just push
+material to your GitHub repository with `git add`, `git commit`, and
+`git push`.
+
+The sites use [Jekyll](https://jekyllrb.com/), a
+[ruby](https://www.ruby-lang.org/en/) [gem](https://rubygems.org/), to
+convert Markdown files to html, and this part is done
+automatically when you push the materials to the `gh-pages` branch
+of a GitHub repository.
```
Comment (AlexBuccheri) [inline docs/index.md:20]: Note to self, this should be removed/replaced. It was just a placeholder to impact so info on Github pages

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Components
+
+- [Minimax Time-Frequency](gx_time_frequency.md)
+
+## Installation (?)
```
Comment (AlexBuccheri) [inline docs/index.md:26]: This heading should be removed while there's no installation details.

File: docs/gx_time_frequency.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+# Usage
+
+# Benchmarks
+
+## CH<sub>4</sub> RPA
+
+In this test, we evaluate the RPA total energy of CH4 using a Gauss-Legendre grid, a modified Gauss-Legendre grid (so far the standard in FHI-aims and abinit), and minimax grids. An accuracy of 10^-6 eV is reached with 10 minimax grid points while the modified Gauss-Legendre grids requires 36 points for this accuracy.
+
+[CH4 benchmark](./img/ch4_becnh.png)
```
Comment (AlexBuccheri) [inline docs/gx_time_frequency.md:18]: `ch4_becnh.png` should be renamed `ch4_bench.png`

**Comment by AlexBuccheri** [comment]
@Panadestein 

> Hey @AlexBuccheri I have added a section with the CH4 benchmark (verbatim from the issue, it is just md). Now for the action, I think GitHub deals with that alone, you just have to enable it in the Settings of the repo. Since we have only md files, Jekyll will do that just fine. I have some actions for my websites, but only because I use other formats and I convert the files to HTML (basically the same as Jekyll does).

It's enabled. I assume we see deployment when the branch gets merged? (my first time setting this up)

**Comment by AlexBuccheri** [comment]
As this appears to be readying completion, I'll get the blurb done today

File: docs/index.md (hunk 4 above)
Comment (Panadestein) [reply docs/index.md:26]: Sure

File: docs/gx_time_frequency.md (hunk 5 above)
Comment (Panadestein) [reply docs/gx_time_frequency.md:18]: Done

**Comment by Panadestein** [comment]
@AlexBuccheri should we merge this PR and see how the website looks? Then we create another PR

File: JOSS/paper.md (hunk 1 above)
Comment (AlexBuccheri) [reply JOSS/paper.md:168]: I will leave it for the time-being

**Comment by Panadestein** [comment]
Cool, the site is alive https://nomad-coe.github.io/greenX/

Let's add it to the description of the repo, top right :))

**Pull request merged**